Add es domain encryption support

garethmdavies opened the pull request:
Attempt to add encryption support to ES domain. Consideration: should the encryption be enabled by default?

garethmdavies committed:
Add encryption at rest variable

garethmdavies committed:
Add encryption attributes to es domain with KMS resource.

garethmdavies committed:
Make KMS resources conditional on encryption variables.

mattops commented on main.tf:
```diff
@@ -183,12 +183,37 @@ resource "null_resource" "es_ns_annotation" {
 }
 
+resource "aws_kms_key" "kms" {
+  count       = var.encryption_at_rest == "true" || var.node_to_node_encryption_enabled == "true" ? 1 : 0
```
Do you need the 
```
|| var.node_to_node_encryption_enabled == "true"
```
Not sure that node_to_node_encryption requires the KMS key.

garethmdavies committed:
Remove the unnecessary node to node encryption check

digitalronin committed:
Describe `encryption_at_rest` in README

This adds `encryption_at_rest` to the list of variables in the table in
the README.

garethmdavies committed:
Fix the encryption properties syntax

garethmdavies committed:
Merge branch 'add-es-domain-encryption-support' of https://github.com/ministryofjustice/cloud-platform-terraform-elasticsearch into add-es-domain-encryption-support

mogaal commented on variables.tf:
```diff
@@ -164,6 +164,13 @@ variable "node_to_node_encryption_enabled" {
 }
 
+variable "encryption_at_rest" {
+  type        = string
```
Hi. Thanks for your contribution. Can you please change this to type bool, terraform support bool type.

mogaal commented on main.tf:
```diff
@@ -183,12 +183,42 @@ resource "null_resource" "es_ns_annotation" {
 }
 
+resource "aws_kms_key" "kms" {
+  count       = var.encryption_at_rest == "true" ? 1 : 0
```
Once you change variable type to `bool` this can be written like

```
  count       = var.encryption_at_rest ? 1 : 0
```

mogaal commented on main.tf:
```diff
@@ -183,12 +183,42 @@ resource "null_resource" "es_ns_annotation" {
+}
+
+resource "aws_kms_alias" "alias" {
+  count       = var.encryption_at_rest == "true" ? 1 : 0
```
Same here, once you change variable type to `bool` this can be written like

```
  count       = var.encryption_at_rest ? 1 : 0
```

mogaal commented on main.tf:
```diff
@@ -183,12 +183,42 @@ resource "null_resource" "es_ns_annotation" {
 
+  encrypt_at_rest {
+    enabled = var.encryption_at_rest
+    kms_key_id = var.encryption_at_rest == "true" ? aws_kms_key.kms[0].arn : null
```
Also here.

mattops committed:
Switch to using bool type for new parameters
